engine/samples/imgui/main.cpp

```diff
@@ -1,12 +1,92 @@
-/// [Full code]
 /// [Including plugin headers]
+#include <iostream>
+#include <map>
+#include <vector>
+
 #include <imgui.h>
 
+#include <cubos/core/reflection/external/map.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+#include <cubos/core/reflection/external/string.hpp>
+#include <cubos/core/reflection/external/vector.hpp>
+#include <cubos/core/reflection/reflect.hpp>
+#include <cubos/core/reflection/traits/fields.hpp>
+
+#include <cubos/engine/imgui/data_inspector.hpp>
 #include <cubos/engine/imgui/plugin.hpp>
+
 /// [Including plugin headers]
 
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+using cubos::core::reflection::reflect;
+using cubos::engine::DataInspector;
 using namespace cubos::engine;
 
+/// [Creating a dummy resource]
+struct Person
+{
+    CUBOS_REFLECT;
+    std::string name;
+    int32_t age;
+    float weight;
+    bool dead;
+};
+
+CUBOS_REFLECT_IMPL(Person)
+{
+    return cubos::core::reflection::Type::create("Person").with(cubos::core::reflection::FieldsTrait()
+                                                                    .withField("name", &Person::name)
+                                                                    .withField("age", &Person::age)
+                                                                    .withField("weight", &Person::weight)
+                                                                    .withField("dead", &Person::dead));
+}
+
+struct DummyResource
+{
+    int integer;
+    Person person;
+    std::vector<int32_t> vec;
+    std::map<int32_t, int32_t> map;
+};
+/// [Creating a dummy resource]
+
+/// [Filling the dummy resource 1]
+static void fillDumyResourceSystem(Write<DummyResource> data)
+{
+    data->integer = 1337;
+    data->person = Person{"roby", 1337, 666.5F, false};
+    data->vec = {12, 59, 25};
+    data->map = {
+        {1, 2},
+        {2, 4},
+        {3, 6},
+        {4, 8},
+    };
+}
+/// [Filling the dummy resource 1]
+
+/// [DataInspector window example]
+static void exampleDataInspectorSystem(Write<DataInspector> inspector, Read<DummyResource> data)
+{
+    static const auto& integerType = reflect<int>();
+    static const auto& personType = reflect<Person>();
+    static const auto& vecType = reflect<std::vector<int32_t>>();
+    static const auto& mapType = reflect<std::map<int32_t, int32_t>>();
+
+    ImGui::Begin("Data Inspector");
+    ImGui::BeginTable("id1", 2);
+
+    inspector->inspect("data->integer", integerType, (void*)&data->integer);
+    inspector->inspect("data->person", personType, (void*)&data->person, false);
+    inspector->inspect("data->vec", vecType, (void*)&data->vec, false);
+    inspector->inspect("data->map", mapType, (void*)&data->map, false);
```

Comment:
If you add the templated show/edit methods, you no longer need those static vars above